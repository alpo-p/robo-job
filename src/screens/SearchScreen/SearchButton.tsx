import { useTheme } from '@react-navigation/native'
import React from 'react'
import { TouchableOpacity } from 'react-native'
import RoboText from '../../common/components/RoboText'

export const SearchButton: React.FC<ButtonProps> = ({ onPress }) => {
  const { colors } = useTheme()
  return (
    <TouchableOpacity
      onPress={onPress}
      style={{
        alignSelf: 'center',
        justifyContent: 'center',
        width: '91%',
        backgroundColor: colors.primary,
        borderRadius: 8 * 1,
        alignItems: 'center',
        height: 8 * 5,
      }}
    >
      <RoboText size="small" weight="bold" color="white" uppercase>
        Search
      </RoboText>
    </TouchableOpacity>
  )
}
interface ButtonProps {
  onPress: () => void
}
