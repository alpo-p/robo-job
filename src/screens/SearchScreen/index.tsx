import { useNavigation } from '@react-navigation/native'
import React, { useState } from 'react'
import { SafeAreaView } from 'react-native-safe-area-context'
import { HomeNavigationPropType } from '../../navigators/HomeScreenNavigator'
import InputBarWithIcon from '../../common/InputBarWithIcon'
import { SearchButton } from './components/SearchButton'
import SearchTags from './components/SearchTags'
import { getRandomId } from '../../utils/utils'
import { TagType } from '../../common/types'
import waitSeconds from '../../utils/waitSeconds'

export default () => {
  const navigation = useNavigation<HomeNavigationPropType>()
  const [search, setSearch] = useState('')
  const [searchTags, setSearchTags] = useState<TagType[]>([])
  const [location, setLocation] = useState('')
  const [locationTags, setLocationTags] = useState<TagType[]>([])

  const handleSearch = () => navigation.navigate('HomeScreen')

  const onEndEditingSearch = () => {
    if (search) {
      const tag: TagType = {
        id: getRandomId(),
        text: search,
      }
      setSearchTags(s => s.concat(tag))
      setSearch('')
    }
  }

  const onEndEditingLocation = () => {
    if (location) {
      const tag: TagType = {
        id: getRandomId(),
        text: location,
      }
      setLocationTags(l => l.concat(tag))
      setLocation('')
    }
  }

  const handleRemoveSearchTag = async (id: string) => {
    // Looks cooler this way
    await waitSeconds(0.2)
    setSearchTags(s => s.filter(tag => tag.id !== id))
  }

  const handleRemoveLocationTag = async (id: string) => {
    await waitSeconds(0.2)
    setLocationTags(l => l.filter(tag => tag.id !== id))
  }

  return (
    <SafeAreaView style={{ marginTop: '50%' }}>
      <InputBarWithIcon
        value={search}
        setValue={setSearch}
        iconName="search"
        onEndEditing={onEndEditingSearch}
      />
      <SearchTags tags={searchTags} onPressTag={handleRemoveSearchTag} />
      <InputBarWithIcon
        value={location}
        setValue={setLocation}
        iconName="location"
        placeholderText="Location"
        onEndEditing={onEndEditingLocation}
      />
      <SearchTags tags={locationTags} onPressTag={handleRemoveLocationTag} />
      <SearchButton onPress={handleSearch} />
    </SafeAreaView>
  )
}
