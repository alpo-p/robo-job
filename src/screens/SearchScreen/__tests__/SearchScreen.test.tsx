import React from 'react'
import create from '../../../testUtils/rendererCreateWithNavigation'
import SearchScreen from '../index'

describe('SearchScreen', () => {
  it('should match snapshot', () => {
    const tree = create(<SearchScreen />)
    expect(tree).toMatchSnapshot()
  })
})
