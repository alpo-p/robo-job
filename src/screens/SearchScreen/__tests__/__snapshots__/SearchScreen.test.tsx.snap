// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchScreen should match snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "marginTop": "65%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#f1f1f1",
          "borderRadius": 8,
          "flexDirection": "row",
          "height": 32,
          "margin": 16,
        },
        Object {
          "marginBottom": 8,
        },
      ]
    }
  >
    <Text />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      placeholder="Search"
      placeholderTextColor="rgba(28, 28, 30, 0.4)"
      rejectResponderTermination={true}
      returnKeyType="done"
      style={
        Object {
          "color": "#000000",
          "width": "70%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginLeft": 16,
      }
    }
    horizontal={true}
    showsHorizontalScrollIndicator={false}
  >
    <View />
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#f1f1f1",
          "borderRadius": 8,
          "flexDirection": "row",
          "height": 32,
          "margin": 16,
        },
        Object {
          "marginBottom": 8,
        },
      ]
    }
  >
    <Text />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      placeholder="Location"
      placeholderTextColor="rgba(28, 28, 30, 0.4)"
      rejectResponderTermination={true}
      returnKeyType="done"
      style={
        Object {
          "color": "#000000",
          "width": "70%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginLeft": 16,
      }
    }
    horizontal={true}
    showsHorizontalScrollIndicator={false}
  >
    <View />
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#000000",
        "borderRadius": 8,
        "height": 40,
        "justifyContent": "center",
        "marginTop": 16,
        "opacity": 1,
        "width": "91.5%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "DamascusBold",
          "fontSize": 14,
          "textTransform": "uppercase",
        }
      }
    >
      Search
    </Text>
  </View>
  <View
    style={
      Object {
        "marginLeft": 16,
        "marginTop": 80,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Damascus",
          "fontSize": 20,
          "textTransform": "none",
        }
      }
    >
      Recent searches
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 8,
        }
      }
    >
      <RCTScrollView
        horizontal={true}
      >
        <View>
          <View
            style={
              Object {
                "alignContent": "center",
                "borderColor": "#000000",
                "borderRadius": 4,
                "borderWidth": 0.5,
                "flexDirection": "row",
                "height": 24,
                "marginRight": 6,
                "marginTop": 8,
                "padding": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "DamascusLight",
                  "fontSize": 12,
                  "textTransform": "uppercase",
                }
              }
            >
              #
               
              marketing
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "borderColor": "#000000",
                "borderRadius": 4,
                "borderWidth": 0.5,
                "flexDirection": "row",
                "height": 24,
                "marginRight": 6,
                "marginTop": 8,
                "padding": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "DamascusLight",
                  "fontSize": 12,
                  "textTransform": "uppercase",
                }
              }
            >
              #
               
              social media
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "borderColor": "#000000",
                "borderRadius": 4,
                "borderWidth": 0.5,
                "flexDirection": "row",
                "height": 24,
                "marginRight": 6,
                "marginTop": 8,
                "padding": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "DamascusLight",
                  "fontSize": 12,
                  "textTransform": "uppercase",
                }
              }
            >
              #
               
              Helsinki
            </Text>
          </View>
        </View>
      </RCTScrollView>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#000000",
            "opacity": 1,
            "position": "absolute",
            "right": 16,
            "top": 3,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 8,
        }
      }
    >
      <RCTScrollView
        horizontal={true}
      >
        <View>
          <View
            style={
              Object {
                "alignContent": "center",
                "borderColor": "#000000",
                "borderRadius": 4,
                "borderWidth": 0.5,
                "flexDirection": "row",
                "height": 24,
                "marginRight": 6,
                "marginTop": 8,
                "padding": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "DamascusLight",
                  "fontSize": 12,
                  "textTransform": "uppercase",
                }
              }
            >
              #
               
              marketing automation
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "borderColor": "#000000",
                "borderRadius": 4,
                "borderWidth": 0.5,
                "flexDirection": "row",
                "height": 24,
                "marginRight": 6,
                "marginTop": 8,
                "padding": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "DamascusLight",
                  "fontSize": 12,
                  "textTransform": "uppercase",
                }
              }
            >
              #
               
              Remote
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "borderColor": "#000000",
                "borderRadius": 4,
                "borderWidth": 0.5,
                "flexDirection": "row",
                "height": 24,
                "marginRight": 6,
                "marginTop": 8,
                "padding": 4,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "DamascusLight",
                  "fontSize": 12,
                  "textTransform": "uppercase",
                }
              }
            >
              #
               
              Helsinki
            </Text>
          </View>
        </View>
      </RCTScrollView>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#000000",
            "opacity": 1,
            "position": "absolute",
            "right": 16,
            "top": 3,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
